app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -0,0 +1,116 @@
+module ForemanAzureRM
+  class AzureRMCompute
+    attr_accessor :sdk
+    attr_accessor :azure_vm
+
+    delegate :id, :name, to: :azure_vm, allow_nil: true
+
+    def initialize(azure_vm: ComputeModels::VirtualMachine.new, sdk: sdk)
+      @azure_vm = azure_vm
+      @sdk = sdk
+    end
+
+    # All the earlier properties are no longer present here, they've
+    # been wrapped directly inside view.
+    # Only foreman related properties would be taken care of here.
+    # id and name have been shifted to delegate.
+
+
+    def persisted?
+      !!identity && !!id
+    end
+
+    def vm_size
+      @azure_vm.hardware_profile.vm_size
+    end
+
+    def wait_for(_timeout = 0, _interval = 0, &block)
+      instance_eval(&block)
+      return true
+    end
+
+    def ready?
+      vm_status == 'running'
+    end
+
+    def state
+      vm_status
+    end
+
+    def vm_status
+      # 2 calls.
+      # 1st works with @azure_vm.resource_group since before
+      # Setup Compute Instance step in orchestration and during the
+      # orchestration steps @azure_vm is the object of
+      # class "Azure::Compute::Mgmt::V2019_03_01::Models::VirtualMachine"
+      # 2nd works with @azure_vm.azure_vm.resource_group (when @azure_vm
+      # becomes the object of class "ForemanAzureRM::AzureRMCompute".)
+
+      require 'pry';binding.pry
+      sdk.check_vm_status(@azure_vm.resource_group, name)
```

Comment:
The `vm_status` method is called once during the `setComputeIP` step in orchestration and secondly once the vm is provisioned when we show the vm details.
Now that we have [properties under :azure_vm(using fields_for)](https://github.com/theforeman/foreman_azure_rm/pull/36/files#diff-c87d7c3c0dfabd338e688a68a3f0dde5R91), this vm_status method fails during the second call and instead will be accessible in the form: `@azure_vm.azure_vm.resource_group`. 
So, how should we approach to unify this?

Reply:
Now that we are sure we do not wrap the wrapper, it should not happen. `@azure_vm.resource_group` is the correct implementation.